packages/template-retail-react-app/app/pages/registration/index.jsx

```diff
@@ -7,36 +7,53 @@
 
 import React, {useEffect} from 'react'
 import PropTypes from 'prop-types'
+import {useIntl} from 'react-intl'
 import {Box, Container} from '@chakra-ui/react'
-import useCustomer from '../../commerce-api/hooks/useCustomer'
-import Seo from '../../components/seo'
+import {
+    ShopperLoginHelpers,
+    useShopperLoginHelper,
+    useCustomerType
+} from 'commerce-sdk-react-preview'
 import {useForm} from 'react-hook-form'
+import {useLocation} from 'react-router-dom'
+import Seo from '../../components/seo'
 import RegisterForm from '../../components/register'
 import useNavigation from '../../hooks/use-navigation'
 import useEinstein from '../../commerce-api/hooks/useEinstein'
-import {useLocation} from 'react-router-dom'
+import {API_ERROR_MESSAGE} from '../../constants'
 
 const Registration = () => {
+    const {formatMessage} = useIntl()
     const navigate = useNavigation()
-    const customer = useCustomer()
+    const {isRegistered} = useCustomerType()
     const form = useForm()
     const einstein = useEinstein()
     const {pathname} = useLocation()
+    const register = useShopperLoginHelper(ShopperLoginHelpers.Register)
 
-    const submitForm = async (data) => {
-        try {
-            await customer.registerCustomer(data)
-        } catch (error) {
-            form.setError('global', {type: 'manual', message: error.message})
+    const submitForm = (data) => {
+        const body = {
+            customer: {
+                firstName: data.firstName,
+                lastName: data.lastName,
+                email: data.email,
+                login: data.email
+            },
+            password: data.password
         }
+
+        return register.mutateAsync(body, {
+            onError: () => {
+                form.setError('global', {type: 'manual', message: formatMessage(API_ERROR_MESSAGE)})
+            }
```

Comment:
I know this is outside the scope of this PR. But it would be nice for use to handle more of the actual errors. Maybe we can leave a note here to do that. 

Here is a list of the possible errors coming back from the server: 

```


400 | CustomerAlreadyRegisteredException |   | Indicates that the resource is called with JWT representing a registered customer.
-- | -- | -- | --
400 | MissingEmailException |   | Indicates that request document does not contain email.
400 | MissingLastNameException |   | Indicates that request document does not contain last_name.
400 | MissingLoginException |   | Indicates that request document does not contain login.
400 | LoginAlreadyInUseException |   | Indicates that the given login is already used.
400 | InvalidLoginException |   | Indicates that login doesn't match acceptance criteria.
400 | InvalidPasswordException |   | Indicates that password doesn't match acceptance criteria.
400 | MissingPasswordException |   | Indicates that password was not provided in JWT scenario.


```